src/panels/lovelace/elements/hui-state-label-element.ts

```diff
@@ -58,6 +59,25 @@ class HuiStateLabelElement extends LitElement implements LovelaceElement {
       `;
     }
 
+    if (attribute && !stateObj.attributes[attribute]) {
+      return html`
+        <hui-warning-element
+          label=${this.hass.localize(
+            "ui.panel.lovelace.warning.attribute_not_found",
+            "attribute",
+            this._config.attribute,
+            "entity",
+            this._config.entity
+          )}
+        ></hui-warning-element>
+      `;
+    }
+
+    const stateDisplay = stateObj
+      ? computeStateDisplay(this.hass.localize, stateObj, this.hass.language)
+      : "-";
```

Comment:
Not sure I see why you are computing the display here? Just inline it in the HTML template

Reply:
I didn't want nested ternaries (they are hard to read), so I pre-computed the values here

Reply:
hmm maybe I am missing something but at this point in the code (and even the old code) is stateObj ever undefined? We have already checked for it in the above IF and return if it is undefined

Reply:
You're not. stateObj is already checked